src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSelectChange } from '@angular/material/select';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { TDataState } from 'src/app/common/http/common.http.types';
+import { SystemNotificationsService } from 'src/app/common/utils/system-notifications/system-notifications.service';
+import { PagesWalletsManagementService } from '../management/pages-wallets-management.service';
+import { WalletsManagementItem } from '../management/pages-wallets-wallets-management-item.model';
+import { PagesWalletsManagementEditorComponent } from '../management/wallet-editor/pages-wallets-management-editor.component';
+import { IWalletModalData } from '../management/wallet-editor/pages-wallets-management-editor.types';
+import { WalletsDetailsTransaction } from './pages-wallet-details-item.model';
+import { PagesWalletDetailsService } from './pages-wallet-details.service';
+import { WalletTransactionType } from './pages-wallet-details.types';
+
+@Component({
+  selector: 'app-pages-wallet-details',
+  templateUrl: './pages-wallet-details.component.html',
+  styleUrls: ['./pages-wallet-details.component.scss'],
+})
+export class PagesWalletDetailsComponent implements OnInit {
+
+  toppings = new FormControl('');
+
+  public walletTransactionType: typeof WalletTransactionType = WalletTransactionType;
+
+  public selectedTransactionType: WalletTransactionType = WalletTransactionType.AllTransaction;
+
+  public displayedColumns: string[] = ['date', 'description', 'amount', 'actions'];
+
+  public walletsManagementItem?: WalletsManagementItem;
+
+  public displayedTransactions: WalletsDetailsTransaction[] = [];
+
+  public walletsDetailsData: TDataState<WalletsDetailsTransaction[]> = {
+    data: null,
+    hasError: false,
+    isLoading: true,
+  }
+
+  public constructor(
+    private readonly pagesWalletDetailsService: PagesWalletDetailsService,
+    private readonly myWalletsService: PagesWalletsManagementService,
+    private readonly systemNotificationsService: SystemNotificationsService,
+    private readonly dialog: MatDialog,
+    private readonly activatedRoute: ActivatedRoute,
+  ) { }
+
+  public ngOnInit(): void {
+    this.activatedRoute.data.subscribe( ({ wallet }) => {
+      this.walletsManagementItem = wallet;
+      this.initWalletDetails(wallet);
+    });
+
+    this.toppings.valueChanges.subscribe((data) => {
```

Comment:
When subscribing you need to always remember about unsubscribing to avoid memory leaks https://levelup.gitconnected.com/preventing-angular-subscription-memory-leaks-65f18eb9cb8e